verifier/src/lib.rs

```diff
@@ -17,8 +17,16 @@ pub use winterfell::StarkProof;
 /// Returns Ok(()) if the specified program was executed correctly against the specified inputs
 /// and outputs.
 ///
-/// Specifically, verifies that if a program with the specified `program_hash` is executed with the
-/// provided `public_inputs` and some secret inputs, and the result is equal to the `outputs`.
+/// Specifically, verifies that if a program with the specified `program_hash` is executed against
+/// the provided `stack_inputs` and some secret inputs, the result is equal to the `stack_outputs`.
+///
+/// Stack inputs are expected to be provided in the order as if they were pushed onto the stack
+/// one by one. Thus, the last value in the `stack_inputs` slice is expected to be the value at
+/// the top of the stack.
```

Comment:
I find this a bit unclear. I think it's because the verifier won't actually be pushing items onto the stack, which makes it harder for us to use straightforward tenses, but maybe we shouldn't worry about that detail so much here? Here's an attempted rephrase (and feel free to adjust)

"Stack inputs are expected to be ordered as if they would be pushed onto the stack one by one. Thus, their expected order on the stack will be the reverse of the order in which they are provided, and the last value in the `stack_inputs` slice is expected to be the value at the top of the stack."

Reply:
I like your version better. Updated!